process.go

```diff
@@ -159,28 +135,37 @@ func getSeekInfo(file *os.File, fromLineNumber int) tail.SeekInfo {
 	}
 }
 
-func processFile(fileInput *FileInputConfig, g *grok.Grok) {
-	defer wg.Done()
-
+func processFile(fileInput *FileInputConfig, g *grok.Grok, cond *sync.Cond) {
 	absFilePath, err := filepath.Abs(fileInput.File)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal().Err(err).Msgf("Getting absolute file path of %q", fileInput.File)
 	}
+
+	log.Debug().Str("path", absFilePath).Msg("Opening file")
+
 	file, err := os.Open(absFilePath)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatal().Err(err).Msg("Opening file")
 	}
-	defer file.Close()
+	defer func() {
+		log.Debug().Str("path", absFilePath).Msg("Closing file")
+
+		err := file.Close()
+		if err != nil {
+			log.Error().Err(err).Str("path", absFilePath).Msg("Closing file")
+		}
+	}()
```

Comment:
I just realized that we never close the file. Now it should not be a problem anymore.